internal/bminventory/inventory.go

```diff
@@ -3186,3 +3196,85 @@ func proxySettingsForIgnition(httpProxy, httpsProxy, noProxy string) (string, er
 	}
 	return buf.String(), nil
 }
+
+func (b *bareMetalInventory) RegisterOCPCluster(ctx context.Context) error {
+	log := logutil.FromContext(ctx, b.log)
+	id := strfmt.UUID(uuid.New().String())
+	url := installer.GetClusterURL{ClusterID: id}
+	clusterName := "ocp-assisted-service-cluster"
+
+	log.Infof("Register OCP cluster: %s with id %s", clusterName, id.String())
+
+	apiVIP, err := b.getApiVIPFromOCP(log)
+	if err != nil {
+		return err
+	}
+
+	openshiftVersion, err := b.getOpenshiftVersionFromOCP(log)
+	if err != nil {
+		return err
+	}
+
+	cluster := common.Cluster{Cluster: models.Cluster{
+		ID:               &id,
+		Href:             swag.String(url.String()),
+		Kind:             swag.String(models.ClusterKindAddHostsOCPCluster),
+		Name:             clusterName,
+		OpenshiftVersion: openshiftVersion,
+		UserName:         auth.UserNameFromContext(ctx),
+		OrgID:            auth.OrgIDFromContext(ctx),
+		UpdatedAt:        strfmt.DateTime{},
+		APIVipDNSName:    &apiVIP,
+	}}
+
+	err = validations.ValidateClusterNameFormat(clusterName)
+	if err != nil {
+		log.Errorf("Failed to validate cluster name: %s", clusterName)
+		return err
+	}
+
+	// Persist worker-ignition to s3 for cluster
+	err = b.createAndUploadNodeIgnition(ctx, &id, apiVIP)
+	if err != nil {
+		log.Errorf("Failed to create and upload worker ignition for cluster %s", id)
+		return err
+	}
+
+	// After registering the cluster, its status should be 'ClusterStatusAddingHosts'
+	err = b.clusterApi.RegisterAddHostsCluster(ctx, &cluster)
+	if err != nil {
+		log.Errorf("failed to register cluster %s ", clusterName)
+		return err
+	}
+
+	return nil
+}
+
+func (b *bareMetalInventory) getApiVIPFromOCP(log logrus.FieldLogger) (string, error) {
+	configMap, err := b.k8sClient.GetConfigMap("kube-system", "cluster-config-v1")
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get configmap cluster-config-v1 from namespace kube-system")
+		return "", err
+	}
+	configStruct := make(map[string]interface{})
+	err = yaml.Unmarshal([]byte(configMap.Data["install-config"]), configStruct)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to unmarshal confimap cluster-config-v1 data: <%s>", configMap.Data["install-config"])
+		return "", err
+	}
+	platform := configStruct["platform"].(map[interface{}]interface{})
```

Comment:
Do you need to check that it's not nil? 
you will have panic if something is not set correctly.

Reply:
fixed